src/main/java/net/minecraftforge/gradle/tasks/ObtainFernFlowerTask.java

```diff
@@ -27,14 +27,19 @@ public class ObtainFernFlowerTask extends CachedTask {
     @OutputFile
     private DelayedFile ffJar;
 
-    @TaskAction
-    public void doTask() throws IOException {
+    public ObtainFernFlowerTask() {
         if (getProject().getGradle().getStartParameter().isOffline()) {
             getLogger().error("Offline mode! not downloading Fernflower!");
```

Comment:
I think the Offline Mode error log should be shown only if this is executed instead of registering this task.
If obtaining is not required, this task will be skipped & the log should not be shown, I think.